### ISSUE TITLE
Expenses-reports: Expenses cannot be listed by payment-date

### ISSUE THREAD

**Issue by n1smithy** [invoiceninja/admin-portal#646]
## Setup
- Version: v5.8.49-C157
- Environment: Docker

## Interface
- Flutter: []

## Checklist
- Can you replicate the issue on our v5 demo site https://demo.invoiceninja.com or https://react.invoicing.co/demo? Demo-site is not available right now :(
- Have you searched existing issues? Yes
- Have you reported this to Slack/forum before posting? Forum
- Have you inspected the logs in storage/logs/laravel.log for any errors? No

## Describe the bug
Payments are only sorted or displayed by date, not by payment-date (!) even when this option is shown.

### Steps To Reproduce
Create an expense by date 28th Jan. 2024. Set it to paid by payment-date of 2nd March 2024. Try to create a report for all expenses paid in March 24. The expense won't be shown.

### Expected Behavior
Expenses should be listed by the right payment-date, not only by the (expense-) date.

**Comment by hillelcoren** [comment]
Hi,

Have you tried adding the expense's payment date column to the report to filter the data?

**Comment by n1smithy** [comment]
> Hi,
> 
> Have you tried adding the expense's payment date column to the report to filter the data?

Hi, Hillel,

that would be the second step before the first, because if one has expenses which were set up months earlier one had to take the whole bunch of expenses just to filter 3 months or less out. And then, as in my case, if one need a seperation between the tax-rates, there is no further filter-option.

**Comment by hillelcoren** [comment]
I'm not sure I follow, the app supports filtering the reports by more than one column. 

Are you using the reports or the export option on Settings > Import/Export?

**Comment by n1smithy** [comment]
Reports in the Flutter-app to check the values.

**Comment by hillelcoren** [comment]
Can you please help me understand your request?

**Comment by n1smithy** [comment]
I set up a report with following settings:
Report: Expenses, Group: VAT-Group 1 (in German: Steuersatz 1), Date: Payment-date (!), Period: Last quarter

The problem is: expenses which were invoiced already one month (or maybe more) before the payment-date are not shown, because there's a bug (or maybe a not intuitive behavior) within the date-option/-filter.

**Comment by n1smithy** [comment]
The demo.invoiceninja.com is not working; otherwise I had taken some screenshots

**Comment by hillelcoren** [comment]
Ok, so the request is to be able to filter by multiple dates?

**Comment by n1smithy** [comment]
No, only (or in first instance) by payment-date (not by expense-date)

**Comment by n1smithy** [comment]
In the Flutter-web-app there are two possibilities to chose: expense-date or payment-date. But payment-date still filters (first) by the expense-date, so expenses which have a earlier expense-date are from the start on not involved even if their payment-date fits into the chosen period.

**Comment by hillelcoren** [comment]
Each column should only filter it's respective value, if you filter by the payment date it shouldn't consider the expense date. 

We'll work on fixing the demo, I suggest posting a screenshot when it's back online.

**Comment by n1smithy** [comment]
Maybe "filter" is the wrong term. Before filtering an output one has to define the data which shall be shown in the output (like the output of a database-query). So - for example - my database-query is something like:

`SELECT _payment_**date**_ (NOT expense_date or similar), payment_amount, tax_rate, tax_amount, total_amount (, …) FROM expenses;`

One can set the date-value as date or payment-date in the Flutter-App at the top like in the following screenshot (top-right setting). But the problem is that in "Bereich" does not completely match "Datum: Zahlungsdatum (payment-date)".
![Bildschirmfoto vom 2024-04-18 09-08-07](https://github.com/invoiceninja/admin-portal/assets/140073251/5eeb896a-1a60-4f82-a396-f381e1cf2192)

Sorry for my bumpy English. It's not my native language so I don't have to improvise sometimes.

**Comment by hillelcoren** [comment]
It sounds like you're trying to filter by two dates but above you said that wasn't the case.

**Comment by n1smithy** [comment]
Nope, I just want an output of all my expenses that I have paid in the defined period. But IN only shows those paid expenses, which expense-date (NOT the payment-date) is within the chosen period.

**Comment by hillelcoren** [comment]
It sounds like you're saying when you set a filter for the payment date it references the expense date? 

I think I'll need a screenshot where the interface is in English with data to understand.
